connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -195,56 +174,69 @@ object KustoWriter {
   def createBlobWriter(schema: StructType,
                        tableCoordinates: KustoCoordinates,
                        tmpTableName: String,
-                       container: CloudBlobContainer): BlobWriteResource = {
-    val blobName = s"${tableCoordinates.database}_${tmpTableName}_${UUID.randomUUID.toString}_SparkStreamUpload.csv.gz"
-    val blob: CloudBlockBlob = container.getBlockBlobReference(blobName)
-    val gzip: GZIPOutputStream = new GZIPOutputStream(blob.openOutputStream())
+                       client: KustoClient): BlobWriteResource = {
+    val blobName = s"${tableCoordinates.database}_${tmpTableName}_${UUID.randomUUID.toString}_spark.csv.gz"
+
+    val containerAndSas = client.getNewTempBlobReference
+    val currentBlob = new CloudBlockBlob(new URI(containerAndSas.containerUrl + '/' + blobName + containerAndSas.sas))
+    val currentSas = containerAndSas.sas
+    val gzip: GZIPOutputStream = new GZIPOutputStream(currentBlob.openOutputStream())
 
     val writer = new OutputStreamWriter(gzip, StandardCharsets.UTF_8)
 
     val buffer: BufferedWriter = new BufferedWriter(writer, GZIP_BUFFER_SIZE)
     val csvWriter = CountingCsvWriter(buffer)
-    BlobWriteResource(buffer, gzip, csvWriter, blob)
+    BlobWriteResource(buffer, gzip, csvWriter, currentBlob, currentSas)
   }
 
   @throws[IOException]
-  private[kusto] def serializeRows(rows: Iterator[InternalRow],
-                                   parameters: KustoWriteResource): Seq[(CloudBlockBlob, Long)] = {
+  private[kusto] def serializeRowsAndIngest(rows: Iterator[InternalRow],
```

Comment:
took the third .. as its cleaner 

i did it inside serialisation because as i serialise i might finish more than one blob - so i can ingest it when its ready

Reply:
Waiting for commit..